src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -142,6 +147,74 @@ public async Task StopContainersAsync(IExecutionContext executionContext, object
             await RemoveContainerNetworkAsync(executionContext, _containerNetwork);
         }
 
+        private async Task<string> GetMSILoginToken(CancellationToken cancellationToken)
+        {
+            // Check environment variable for debugging
+            var envVar = System.Environment.GetEnvironmentVariable("DEBUG_MSI_LOGIN_INFO");
+            // Future: Set this client id. This is the MSI client ID.
+            TokenCredential credential = new ManagedIdentityCredential(clientId: null);
+            if (envVar == "1")
+            {
+                // Future: Set this client id. This is the MSI client ID.
+                credential = new ChainedTokenCredential(new ManagedIdentityCredential(clientId: null), new VisualStudioCredential(), new AzureCliCredential());
+            }
+            var accessToken = await credential.GetTokenAsync(new TokenRequestContext(new[] { "https://management.core.windows.net/" }), cancellationToken);
+            // To print the token, you can convert it to string 
+            return accessToken.Token.ToString();
+        }
+
+        private async Task<string> GetACRPasswordFromAADToken(string AADToken, string tenantId, string registryServer, int retryCount, int timeToWait, CancellationToken cancellationToken)
```

Comment:
Acr, not ACR, see naming conventions: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalization-rules-for-identifiers